nginx.conf.d/http-server-proxy.conf

```diff
@@ -1,8 +1,13 @@
 proxy_http_version 1.1;
-proxy_read_timeout 300s;
 proxy_ssl_server_name on;
 proxy_connect_timeout 30s;
+proxy_read_timeout 300s;
 proxy_set_header Connection "";
+proxy_ignore_headers Cache-Control Vary;
+
+# For POST requests
```

Comment:
just a note, that this buffer can be set also through ingress annotations, which allows us to configure this value for each individual deployment: https://github.com/kubernetes/ingress-nginx/issues/8050#issuecomment-1095735073